EIPS/eip-5005.md

```diff
@@ -0,0 +1,225 @@
+    ) external returns (bool success);
+
+    /// @dev Allows a Module to execute a transaction and return data
+    /// @notice Can only be called by an enabled module.
+    /// @notice Must emit ExecutionFromModuleSuccess(address module) if successful.
+    /// @notice Must emit ExecutionFromModuleFailure(address module) if unsuccessful.
+    /// @param to Destination address of module transaction.
+    /// @param value Ether value of module transaction.
+    /// @param data Data payload of module transaction.
+    /// @param operation Operation type of module transaction: 0 == call, 1 == delegate call.
+    function execTransactionFromModuleReturnData(
+        address to,
+        uint256 value,
+        bytes memory data,
+        Enum.Operation operation
+    ) external returns (bool success, bytes memory returnData);
+
+    /// @dev Returns if an module is enabled
+    /// @return True if the module is enabled
+    function isModuleEnabled(address module) external view returns (bool);
+
+    /// @dev Returns array of modules.
+    /// @param start Start of the page.
+    /// @param pageSize Maximum number of modules that should be returned.
+    /// @return array Array of modules.
+    /// @return next Start of the next page.
+    function getModulesPaginated(address start, uint256 pageSize)
+        external
+        view
+        returns (address[] memory array, address next);
+}
+```
+
+```solidity
+pragma solidity >=0.7.0 <0.9.0;
+
+import "./Enum.sol";
+
+interface IGuard {
+    function checkTransaction(
+        address to,
+        uint256 value,
+        bytes memory data,
+        Enum.Operation operation,
+        uint256 safeTxGas,
+        uint256 baseGas,
+        uint256 gasPrice,
+        address gasToken,
+        address payable refundReceiver,
+        bytes memory signatures,
+        address msgSender
+    ) external;
+
+    function checkAfterExecution(bytes32 txHash, bool success) external;
+}
+
+```
+
+```solidity
+pragma solidity >=0.7.0 <0.9.0;
+
+import "./Enum.sol";
+import "./BaseGuard.sol";
+
+/// @title Guardable - A contract that manages fallback calls made to this contract
+contract Guardable {
+    address public guard;
+
+    event ChangedGuard(address guard);
+
+    /// `guard_` does not implement IERC165.
+    error NotIERC165Compliant(address guard_);
+
+    /// @dev Set a guard that checks transactions before execution.
+    /// @param _guard The address of the guard to be used or the 0 address to disable the guard.
+    function setGuard(address _guard) external {
+        if (_guard != address(0)) {
+            if (!BaseGuard(_guard).supportsInterface(type(IGuard).interfaceId))
+                revert NotIERC165Compliant(_guard);
+        }
+        guard = _guard;
+        emit ChangedGuard(guard);
+    }
+
+    function getGuard() external view returns (address _guard) {
+        return guard;
+    }
+}
+```
+
+```solidity
+// SPDX-License-Identifier: LGPL-3.0-only
```

Comment:
Should be at least compatible with, and ideally, `CC0-1.0`.

Reply:
Removed the license from this snippet as the whole EIP should be CC0.